evalai/utils/requests.py

```diff
@@ -13,73 +12,46 @@
     url = "{}{}".format(get_host_url(), path)
     headers = get_request_header()
+    if method == "POST":
+        files = {"input_file": open(files, "rb")} if files else None
+        data = {"status": "submitting"}
 
-    if method == "GET":
-        try:
-            response = requests.get(url, headers=headers)
-            response.raise_for_status()
-        except requests.exceptions.HTTPError as err:
-            if response.status_code in EVALAI_ERROR_CODES:
-                validate_token(response.json())
-                echo(
-                    style(
-                        "\nError: {}\n".format(response.json().get("error")),
-                        fg="red",
-                        bold=True,
-                    )
-                )
-            else:
-                echo(err)
-            sys.exit(1)
-        except requests.exceptions.RequestException:
+    try:
+        response = requests.request(method, url, data=data, headers=headers, files=files)
+        response.raise_for_status()
+    except requests.exceptions.HTTPError as e:
+        if response.status_code in EVALAI_ERROR_CODES:
+            validate_token(response.json())
             echo(
                 style(
-                    "\nCould not establish a connection to EvalAI."
-                    " Please check the Host URL.\n",
-                    bold=True,
+                    "\nError: {}\n".format(response.json().get("error")),
                     fg="red",
+                    bold=True,
                 )
             )
-            sys.exit(1)
-        return response.json()
-    elif method == "POST":
-        if files:
-            files = {"input_file": open(files, "rb")}
-        else:
-            files = None
-        data = {"status": "submitting"}
-        try:
-            response = requests.post(
-                url, headers=headers, files=files, data=data
-            )
-            response.raise_for_status()
-        except requests.exceptions.HTTPError as err:
-            if response.status_code in EVALAI_ERROR_CODES:
-                validate_token(response.json())
-                echo(
-                    style(
-                        "\nError: {}\n"
-                        "\nUse `evalai challenges` to fetch the active challenges.\n"
-                        "\nUse `evalai challenge CHALLENGE phases` to fetch the "
-                        "active phases.\n".format(response.json()["error"]),
-                        fg="red",
-                        bold=True,
-                    )
-                )
-            else:
-                echo(err)
-            sys.exit(1)
-        except requests.exceptions.RequestException:
             echo(
                 style(
-                    "\nCould not establish a connection to EvalAI."
-                    " Please check the Host URL.\n",
+                    "Use `evalai challenges` to fetch the active challenges."
```

Comment:
@vkartik97 As in, it wasn't there in the code previously. So, it might not make sense to show this message in that case (for example for DELETE method). Also, I'm not sure if we have the support for this in the backend.

Reply:
@vkartik97 It would be better to have some error messages / blocking requests due to security reasons (that can be possible with a custom script, but we can still make a warning block here). For reference: https://www.owasp.org/index.php/Test_HTTP_Methods_(OTG-CONFIG-006)

Reply:
I am thinking of having something like:
```py
if method in [PUT,PATCH, DELETE] then:
    return or/and throw error
else:
.
.
.

```
What are your opinions on this?

Reply:
Great, looks good to me.
I checked the [exceptions list](https://requests.readthedocs.io/en/master/_modules/requests/exceptions/) provided by requests package but it doesn't include an `InvalidMethodError`. So I was thinking we could `echo` a message saying the method is unsupported and then perform a `sys.exit`.
That is:
```python
if method in ["PUT", "PATCH", "DELETE"]:
  echo("Method not supported ...")
  sys.exit(1)
else:
.
.
.
```

Reply:
Instead of `echo` and `exit`, what are opinions on using `raise Exception('{METHOD} not supported by make_request')`? The reason for using exceptions is that the method `make_request` should have a behavior to inform the user/programmer that a exceptional behavior has occurred, here it is `"PUT", "PATCH", "DELETE" is not supported`.
Please let me know if this doesn't seem correct way.

Reply:
@vkartik97 I think that is a valid point. I agree raising an Exception will be better :+1:
I will update the PR with the same.